portal/app/[locale]/tunnel/_components/tunnelProviderToggle.tsx

```diff
@@ -0,0 +1,86 @@
+import { FreeTech } from 'components/customTunnelsThroughPartners/freetech'
+import { Interport } from 'components/customTunnelsThroughPartners/interport'
+import { Meson } from 'components/customTunnelsThroughPartners/meson'
+import { Stargate } from 'components/customTunnelsThroughPartners/stargate'
+import { WarningIcon } from 'components/icons/warningIcon'
+import { useTranslations } from 'next-intl'
+import { getNativeToken } from 'utils/nativeToken'
+import { Chain } from 'viem'
+
+export type TunnelProvider = 'native' | 'thirdParty'
+
+type Props = {
+  fromChainId: Chain['id']
+  onChange: (provider: TunnelProvider) => void
+  provider: TunnelProvider
+  toChainId: Chain['id']
+}
+
+type ThirdPartyOptionsProps = Pick<Props, 'fromChainId' | 'toChainId'>
+
+function ThirdPartyOptions({ fromChainId, toChainId }: ThirdPartyOptionsProps) {
+  const fromNativeToken = getNativeToken(fromChainId)
+  const toNativeToken = getNativeToken(toChainId)
+
+  return (
+    <div className="flex-col items-center justify-center space-y-3 rounded-lg bg-neutral-50 p-4">
+      <Stargate
+        fromToken={fromNativeToken}
+        label="Stargate"
+        toToken={toNativeToken}
+      />
+      <FreeTech />
+      <Interport fromChainId={fromChainId} toChainId={toChainId} />
+      <Meson label="Meson" />
+    </div>
+  )
+}
+
+export const TunnelProviderToggle = function ({
+  fromChainId,
+  onChange,
+  provider,
+  toChainId,
+}: Props) {
+  const t = useTranslations('tunnel-page')
+
+  return (
+    <>
+      <div className="flex w-full items-center justify-center rounded-lg bg-neutral-100 p-1">
+        <button
+          className={`flex-1 rounded-md p-1 text-sm font-medium transition ${
+            provider === 'native'
+              ? 'border-neutral-300/56 border bg-white text-neutral-950 shadow-sm'
+              : 'text-neutral-600'
```

Comment:
If this matches the styles from the Tabs (Which it seems so), there should be a text color update on hover

```suggestion
              : 'text-neutral-600 hover:text-neutral-950'
```

This also applies to the other button a few lines below